PSGSuite/Public/Users/Invoke-GSUserOffboarding.ps1

```diff
@@ -0,0 +1,158 @@
+
+    .PARAMETER License
+    The License to set the user to.
+
+    .EXAMPLE
+    Invoke-GSUserOffboarding -User tom.fields@domain.com -Options Full -DestinationOrgUnit '/Former Employees'
+
+    Performs all of the listed tasks against user Tom Fields, including moving them to the '/Former Employees' OrgUnit and setting them to a VFE license.
+
+    .NOTES
+    Pull requests welcome for functionality enhancements!
+    #>
+    [OutputType('System.String')]
+    [CmdletBinding(SupportsShouldProcess,ConfirmImpact = "High")]
+    Param(
+        [Parameter(Mandatory,Position = 0,ValueFromPipeline,ValueFromPipelineByPropertyName)]
+        [Alias('PrimaryEmail','Mail')]
+        [string[]]
+        $User,
+        [Parameter()]
+        [ValidateSet('Full','ClearASPs','ClearOAuthTokens','RemoveMobileDevices','Suspend','SetRandomPassword','MoveToOrgUnit','SetLicense')]
+        [String[]]
+        $Options = @('ClearASPs','ClearOAuthTokens','RemoveMobileDevices','Suspend','SetRandomPassword'),
+        [Parameter()]
+        [string]
+        $DestinationOrgUnit,
+        [Parameter()]
+        [ValidateSet("G-Suite-Enterprise","Google-Apps-Unlimited","Google-Apps-For-Business","Google-Apps-For-Postini","Google-Apps-Lite","Google-Drive-storage-20GB","Google-Drive-storage-50GB","Google-Drive-storage-200GB","Google-Drive-storage-400GB","Google-Drive-storage-1TB","Google-Drive-storage-2TB","Google-Drive-storage-4TB","Google-Drive-storage-8TB","Google-Drive-storage-16TB","Google-Vault","Google-Vault-Former-Employee","1010020020")]
+        [string]
+        $License = "Google-Vault-Former-Employee"
+    )
+    Begin {
+        function New-RandomPassword {
+            Param (
+                [parameter(Mandatory = $false)]
+                [int]
+                $Length = 15
+            )
+            $ascii = $null
+            for ($a = 33;$a –le 126;$a++) {
+                $ascii += ,[char][byte]$a
+            }
+            for ($loop = 1; $loop –le $length; $loop++) {
+                $randomPassword += ($ascii | Get-Random)
+            }
+            return ([String]$randomPassword)
+        }
+    }
+    Process {
+        foreach ($U in $User) {
+            Resolve-Email ([ref]$U)
+            if ($PSCmdlet.ShouldProcess("Offboarding user: $U")) {
+                Write-Verbose "Offboarding user: $U"
+                "[$(Get-Date -Format o)] Starting offboard of user: $U"
+                $_user = @{User = $U}
+                $updateParams = @{Confirm = $false}
+                foreach ($opt in $options) {
+                    switch -RegEx ($opt) {
+                        '(Full|Suspend)' {
+                            $updateParams['Suspended'] = $true
+                        }
+                        '(Full|SetRandomPassword)' {
+                            $updateParams['Password'] = ConvertTo-SecureString (New-RandomPassword) -AsPlainText -Force
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [File 'Invoke-GSUserOffboarding.ps1' uses ConvertTo-SecureString with plaintext. This will expose secure information. Encrypted standard strings should be used instead.](https://app.codacy.com/manual/scrthq/PSGSuite/pullRequest?prid=4742353)